compiler/src/parser.mly

```diff
@@ -166,6 +171,14 @@ utype:
 | T_U128 { Wsize.U128 }
 | T_U256 { Wsize.U256 }
 
+utype_array:
+| ws=utype {TypeWsize ws}
+| id=ident {TypeSizeAlias id}
+
+utype_cast:
+| ws=utype {TypeWsize ws}
+| id=typealiascall {TypeSizeAlias id}
```

Comment:
Why BACKQUOTE is needed ?

Reply:
In this case the backquote is needed to avoid ambiguity in the grammar. The problem arise from the fact that the parser cannot distinguish a typealias from an expression. We tried several solutions and came to the conclusion that the backquote was the simplest. We could probably avoid it, but it would mean rewriting a large part of the grammar, and probably also modifying pre-typing